classes/blocks/class-gallery.php

```diff
@@ -144,7 +130,7 @@ public function prepare_data( $fields ): array {
 		];
 
 		foreach ( $exploded_images as $image_id ) {
-			$image_size = $image_sizes[ $fields['gallery_block_style'] ];
+			$image_size = $fields['gallery_image_size'] ? $fields['gallery_image_size'] : $image_sizes[ $fields['gallery_block_style'] ];
```

Comment:
If you want you can [shorten this ternary](https://www.php.net/manual/en/language.operators.comparison.php#language.operators.comparison.ternary) to `$image_size = $fields['gallery_image_size'] ?: $image_sizes[ $fields['gallery_block_style'] ];`